src/Monolog/Logger.php

```diff
@@ -148,9 +148,9 @@ class Logger implements LoggerInterface, ResettableInterface
     protected $exceptionHandler;
 
     /**
-     * @var int Keeps track of depth to prevent infinite logging loops
+     * @var int|null Keeps track of depth to prevent infinite logging loops
      */
-    private $logDepth = 0;
+    private $logDepth;
```

Comment:
Explicit null is better as it doesn't break with typed properties (3.x uses typed props and this will merge into it so rather be prepared)

```suggestion
    private $logDepth = null;
```